src/Features/CSharp/Portable/UsePatternCombinators/CSharpUsePatternCombinatorsAnalyzer.cs

```diff
@@ -0,0 +1,203 @@
+                                              op.Syntax is BinaryExpressionSyntax { Right: TypeSyntax type }:
+                    return new Type(type);
+
+                case IIsPatternOperation op when CheckTargetExpression(op.Value) &&
+                                                 op.Pattern.Syntax is PatternSyntax pattern:
+                    return new Source(pattern);
+
+                case IParenthesizedOperation op:
+                    return ParsePattern(op.Operand);
+            }
+
+            return null;
+        }
+
+        private AnalyzedPattern? ParseBinaryPattern(IBinaryOperation op, bool isDisjunctive, SyntaxToken token)
+        {
+            var leftPattern = ParsePattern(op.LeftOperand);
+            if (leftPattern == null)
+                return null;
+
+            var rightPattern = ParsePattern(op.RightOperand);
+            if (rightPattern == null)
+                return null;
+
+            return Binary.Create(leftPattern, rightPattern, isDisjunctive, token);
+
+        }
+
+        private ConstantResult DetermineConstant(IBinaryOperation op)
+        {
+            return (op.LeftOperand, op.RightOperand) switch
+            {
+                var (e, v) when IsConstant(v) && CheckTargetExpression(e) => ConstantResult.Right,
+                var (v, e) when IsConstant(v) && CheckTargetExpression(e) => ConstantResult.Left,
+                _ => ConstantResult.None,
+            };
+        }
+
+        private AnalyzedPattern? ParseRelationalPattern(IBinaryOperation op)
+        {
+            return DetermineConstant(op) switch
+            {
+                ConstantResult.Left when op.LeftOperand.Syntax is ExpressionSyntax left
+                    => new Relational(Flip(op.OperatorKind), left),
+                ConstantResult.Right when op.RightOperand.Syntax is ExpressionSyntax right
+                    => new Relational(op.OperatorKind, right),
+                _ => null
+            };
+        }
+
+        private AnalyzedPattern? ParseConstantPattern(IBinaryOperation op)
+        {
+            return DetermineConstant(op) switch
+            {
+                ConstantResult.Left when op.LeftOperand.Syntax is ExpressionSyntax left
+                    => new Constant(left),
+                ConstantResult.Right when op.RightOperand.Syntax is ExpressionSyntax right
+                    => new Constant(right),
+                _ => null
+            };
+        }
+
+        private static bool IsRelationalOperator(BinaryOperatorKind operatorKind)
+        {
+            switch (operatorKind)
+            {
+                case LessThan:
+                case LessThanOrEqual:
+                case GreaterThanOrEqual:
+                case GreaterThan:
+                    return true;
+                default:
+                    return false;
+            }
+        }
+
+        /// <summary>
+        /// Changes the direction the operator is pointing
+        /// </summary>
```

Comment:
perhaps explain that this is not the negation (i.e. it's not `>   -->     <=`) and why we need this.